src/ui/public/visualize/visualization.js

```diff
@@ -114,16 +121,25 @@ uiModules
           renderSubscription.unsubscribe();
         });
 
-        if (!$scope.vis.visualizeScope) {
```

Comment:
This flag on the `vis` object was set by `visualize` to signal, that we are running inside `visualize`. If we run inside visualize, we don't need to listen on changes in `visData`, `params` or `uiState`, since `visualize` already listens on those, triggers a new data fetch and will then refresh. We only need to listen on those, if we are not running inside `visualize`. This has been replaced by the `listenOnChange` parameter to the `visualization` directive, to make it more explicit and not misuse the `vis` object to carry on information between angular directives.